--- /home/sean/downloads/ps_drone.py	2017-05-24 18:44:02.822231451 -0400
+++ ps_drone.py	2017-05-24 18:10:47.509020022 -0400
@@ -1853,13 +1853,20 @@
 
 ##### ID = 27 ### "zimmu_3000" ################################################
 def decode_ID27(packet):  #NAVDATA_ZIMU_3000_TAG
-	dataset = struct.unpack_from("HHif", packet, offsetND)
-	if dataset[1] != 12 and dataset[1] != 216:		# 216 since firmware 2.4.8 ?
-		print "*** ERROR : navdata-zimmu_3000-Package (ID=27) has the wrong size !!!"
-	zimmu_3000 = [0,0.0]
-	zimmu_3000[0] = dataset[2]	# vzimmuLSB 			   							(int32)
-	zimmu_3000[1] = dataset[3]	# vzfind 			   								(float)
-	return(zimmu_3000)
+	#dataset = struct.unpack_from("HHif", packet, offsetND)
+	#if dataset[1] != 12 and dataset[1] != 216:		# 216 since firmware 2.4.8 ?
+	#	print "*** ERROR : navdata-zimmu_3000-Package (ID=27) has the wrong size !!!"
+	#zimmu_3000 = [0,0.0]
+	#zimmu_3000[0] = dataset[2]	# vzimmuLSB 			   							(int32)
+	#zimmu_3000[1] = dataset[3]	# vzfind 			   								(float)
+	#return(zimmu_3000)
+        ### MANUAL GPS PATCH
+	dataset = struct.unpack_from("IIddddI??ddddIdddddddddddddIdddddI", packet, offsetND)
+        gps = []
+        gps.append(dataset[11])
+        gps.append(dataset[12])
+        gps.append(dataset[13]) #??
+        return(gps)
 
 ##### Footer ### "chksum" #####################################################
 def decode_Footer(packet,allpacket):   ### Decode Checksum options-package ID=65535
@@ -1923,7 +1930,8 @@
 		if dataset[0]==24 and choice[24]: navdata["kalman_pressure"]	= decode_ID24(packet[offsetND:])
 		if dataset[0]==25 and choice[25]: navdata["hdvideo_stream"]		= decode_ID25(packet[offsetND:])
 		if dataset[0]==26 and choice[26]: navdata["wifi"]				= decode_ID26(packet[offsetND:])
-		if dataset[0]==27 and choice[27]: navdata["zimmu_3000"]			= decode_ID27(packet[offsetND:])
+		#if dataset[0]==27 and choice[27]: navdata["zimmu_3000"]			= decode_ID27(packet[offsetND:])
+		if dataset[0]==27 and choice[27]: navdata["gps"]			= decode_ID27(packet[offsetND:])
 		if dataset[0]==65535 and choice[28]: navdata["chksum"]			= decode_Footer(packet[offsetND:],packet)
 		offsetND += dataset[1]
 	return(navdata)
@@ -1945,7 +1953,8 @@
 def mainloopND(DroneIP,NavDataPort,parent_pipe,parentPID):
 	global commitsuicideND
 	something2send, MinimalPacketLength, timetag =	False, 30, 0
-	packetlist =		["demo","time","raw_measures","phys_measures","gyros_offsets","euler_angles","references","trims","rc_references","pwm","altitude","vision_raw","vision_of","vision","vision_perf","trackers_send","vision_detect","watchdog","adc_data_frame","video_stream","games","pressure_raw","magneto","wind_speed","kalman_pressure","hdvideo_stream","wifi","zimmu_3000","chksum","state"]
+	#packetlist =		["demo","time","raw_measures","phys_measures","gyros_offsets","euler_angles","references","trims","rc_references","pwm","altitude","vision_raw","vision_of","vision","vision_perf","trackers_send","vision_detect","watchdog","adc_data_frame","video_stream","games","pressure_raw","magneto","wind_speed","kalman_pressure","hdvideo_stream","wifi","zimmu_3000","chksum","state"]
+	packetlist =            ["demo","time","raw_measures","phys_measures","gyros_offsets","euler_angles","references","trims","rc_references","pwm","altitude","vision_raw","vision_of","vision","vision_perf","trackers_send","vision_detect","watchdog","adc_data_frame","video_stream","games","pressure_raw","magneto","wind_speed","kalman_pressure","hdvideo_stream","wifi","gps","chksum","state"]
 	choice =			[False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,True]
 	overallchoice =		False	# This and oneTimeFailOver is necessary because of a bug (?) of AR.Drone sending NavData in DemoMode...
 	oneTimeFailOver =	True 	# ...while setting a configuration the drone sends the next DemoMode-package with just its status.
